src/test/java/com/thebois/models/beings/BeingTests.java

```diff
@@ -445,84 +440,117 @@ public void allStructuresCompletedNewPlaced() {
         testPawn.update();
 
         // Assert
-        assertThat(testPawn.getFinalDestination().get()).isEqualTo(correctDestination);
+        assertThat(testPawn.getFinalDestination()).isEqualTo(correctDestination);
     }
 
     @Test
-    public void deliverItemsToCloseStructure() {
+    public void pawnDoNotTryToFindNewClosestStructureWhenStillIncomplete() {
         // Arrange
-        final Position positionA = new Position(10, 0);
-        final Position startPosition = new Position(11, 0);
+        final IPathFinder mockPathFinder = Mockito.mock(IPathFinder.class);
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
 
-        final World world = new World(50);
-        final IPathFinder pathFinder = new AstarPathFinder(world);
+        final IStructure mockStructure = Mockito.mock(IStructure.class);
 
-        final Random mockRandom = Mockito.mock(Random.class);
-        when(mockRandom.nextInt(anyInt())).thenReturn(0);
-        when(mockRandom.nextInt(anyInt())).thenReturn(0);
+        final Position expectedPosition = new Position(10f, 20f);
 
-        world.createStructure(positionA);
+        when(mockFinder.findNearestIncompleteStructure(any()))
+            .thenReturn(Optional.of(mockStructure));
 
-        final IStructure structureAtA = world.getStructureAt(positionA).get();
+        when(mockStructure.getPosition()).thenReturn(expectedPosition);
 
-        final Pawn testPawn = new Pawn(startPosition,
-                                       new Position(),
-                                       mockRandom,
-                                       pathFinder,
-                                       world);
+        final Pawn pawn = new Pawn(new Position(),
+                                   new Position(),
+                                   new Random(),
+                                   mockPathFinder,
+                                   mockFinder);
 
         // Act
-        final float initialBuiltRatio = structureAtA.builtStatus();
-        testPawn.update();
+        pawn.update();
+        pawn.update();
```

Comment:
question

Why the double-update?

Reply:
It first needs to find an incomplete structure, then on the second update the pawn should look for a new closest structure since the current one isn't complete